### PR TITLE
Deprecate the remaining helper functions

### PR DESCRIPTION
**Pull request by ausi**
Moved functions:

- `scan()` to `Folder::scan()` 
- `ampersand()` to `StringUtil::ampersand()`
- `array_insert()` to `ArrayUtil::arrayInsert()`
- `array_is_assoc()` to `ArrayUtil::isAssoc()`

Without replacement:

- `nl2br_html5()`
- `nl2br_xhtml()`
- `nl2br_pre()`
- `basename_natcasecmp()`
- `basename_natcasercmp()`
- `natcaseksort()`
- `length_sort_asc()`
- `length_sort_desc()`

### CODE REVIEW

File: core-bundle/src/Resources/contao/library/Contao/ArrayUtil.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	 *
+	 * @return boolean
+	 */
+	public static function isAssoc($arrArray): bool
+	{
+		return \is_array($arrArray) && array_keys($arrArray) !== range(0, \count($arrArray) - 1);
+	}
+}
+
+class_alias(ArrayUtil::class, 'ArrayUtil');
```
Comment (Toflar) [inline core-bundle/src/Resources/contao/library/Contao/ArrayUtil.php]: I don't think that's required? It's new anyway.

**Review by Toflar — CHANGES_REQUESTED** [review]

File: core-bundle/src/Resources/contao/library/Contao/ArrayUtil.php (hunk 1 above)
Comment (ausi) [reply core-bundle/src/Resources/contao/library/Contao/ArrayUtil.php]: Done in b04d31f977ccdb1d1f1fabc25429a12f6d0cec9c

File: core-bundle/src/Resources/contao/library/Contao/Folder.php [hunk 2]
```diff
@@ -539,6 +545,52 @@ protected function getPathinfo()
...
+		}
+
+		// Cache the result
+		if (!$blnUncached)
+		{
+			$arrScanCache[$strFolder] = $arrReturn;
+		}
+
+		return $arrReturn;
+	}
```
Comment (fritzmg) [inline core-bundle/src/Resources/contao/library/Contao/Folder.php:593]: Shouldn't we use the Symfony finder for this instead?

Comment (ausi) [reply core-bundle/src/Resources/contao/library/Contao/Folder.php:593]: Yes, we should. ☺️
But I think this is not required for this pull request. 🙃

**Review by fritzmg — DISMISSED** [review]

**Review by bytehead — DISMISSED** [review]

**Review by leofeyer — APPROVED** [review]

**Review by Toflar — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @ausi.

**Pull request merged**